doc/CMakeLists.txt

```diff
@@ -195,6 +197,17 @@ else()
   set(SEP :)
 endif()
 
+file(WRITE ${KCONFIG_BINARY_DIR}/Kconfig.soc.defconfig
+     "osource \"${ZEPHYR_BASE}/soc/$(ARCH)/*/Kconfig.defconfig\"\n"
+)
+file(WRITE ${KCONFIG_BINARY_DIR}/Kconfig.soc
```

Comment:
Maybe call this one Kconfig.soc_selection, and `s/Kconfig.soc.arch/Kconfig.soc_config/` also?

I'm not finding these file names intuitive.

Reply:
Not sure what `selection` refers to, as we have not yet selected the soc that will be used.

The reason for that names, originates from existing Kconfig names:
https://github.com/zephyrproject-rtos/zephyr/blob/master/Kconfig.zephyr#L25
https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/Kconfig#L6
https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/Kconfig#L11
```
source $(SOC_DIR)/$(ARCH)/*/Kconfig.defconfig  --> Kconfig.soc.defconfig  (defconfig a bit too generic in build folder, so added soc)
source "$(SOC_DIR)/$(ARCH)/*/Kconfig.soc"         --> Kconfig.soc
osource "$(SOC_DIR)/$(ARCH)/Kconfig"                --> Kconfig.soc.arch   (Kconfig.soc already used, and this is soc/arch specific file)
```
We could rename `Kconfig.soc` into `Kconfig.soc.soc`.

> `s/Kconfig.soc.arch/Kconfig.soc_config/`

not sure that name is better. All Kconfig file contains config settings.

Overview of the files and their content (`${SOC_DIR}` removed):
- `$(ARCH)/*/Kconfig.soc`: SOC specific configuration, as example `arm/nordic_nrf/Konfig.soc` loads `arm/nordic_nrf/Konfig.series` which loads nRF52 series specific settings. So this is specific to the soc, not the arch.
  Hence Kconfig.soc is good name in build folder when refering to this file, imho
- `$(ARCH)/Kconfig`: load architecture specific configuration for all SOC in this arch, as example `arm/Kconfig` loads specific configuration settings valid for arm socs, but not for xtensa socs. So specific for all soc belonging to this arch.
    Hence `Kconfig.soc.arch` is good name in build folder when refering to this file, imho
- `$(ARCH)/*/Kconfig.defconfig`: default settings for this soc.
    Hence Kconfig.soc.defconfig is good name when refering to this file, imho

Reply:
Thanks for the justification of the names. No objections.